### PR TITLE
INFENG-5798: Add Dockerfile for building a frugal image

### PR DESCRIPTION
**Pull request by tylerstapler-wf**
We previously created a frugal docker image for people to use when developing, we should add that back.

# Test
```shell
$ docker run --rm -it drydock.workiva.net/workiva/frugal:1334464
Unable to find image 'drydock.workiva.net/workiva/frugal:1334464' locally
1334464: Pulling from workiva/frugal
bd0828aa6cef: Pull complete
Digest: sha256:420bfa702a74017126f93d638a45741e816b64a15121bdf5d364aef8f214c8c0
Status: Downloaded newer image for drydock.workiva.net/workiva/frugal:1334464
Usage: frugal [options] file

Use frugal -help for a list of options
```

@Workiva/infre-product

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

**Comment by charliestrawn-wf** [comment]
QA +1

File: Dockerfile [hunk 1]
```diff
@@ -33,4 +33,30 @@ ARG BUILD_ARTIFACTS_ARTIFACTORY=/go/src/github.com/Workiva/frugal/frugal-*.jar
...
+ARG GIT_SSH_KEY		      
+ARG KNOWN_HOSTS_CONTENT	      
+RUN mkdir /root/.ssh && \     
+  echo "$GIT_SSH_KEY" > /root/.ssh/id_rsa && \
+  echo "$KNOWN_HOSTS_CONTENT" > /root/.ssh/known_hosts && \
+  chmod -R 700 /root/.ssh && \
+  git config --global url."git@github.com:".insteadOf "https://github.com/"
+COPY glide.yaml .
+COPY glide.lock .
+RUN glide install
```
Comment (brianshannan-wf) [inline Dockerfile]: Shouldn't we already have these deps from the previous stage?

Comment (brianshannan-wf) [reply Dockerfile]: Actually, we should be able to use the compiled binary from the previous stage (using the right compilation flags) and skip this stage completely

**Comment by charliestrawn-wf** [comment]
+1

**Comment by charliestrawn-wf** [comment]
QA +1

**Comment by brianshannan-wf** [comment]
+1

@brettkail-wk will this work for what you want?

**Comment by brettkail-wk** [comment]
@brianshannan-wf Yes, verified it works, but it would be nice if the `/bin/frugal` were documented somewhere.

**Comment by tylerstapler-wf** [comment]
@brettkail-wk frugal should be on the container's path. Users can also just pass arguments to the frugal container as a CMD since it's defined as an entrypoint.

What kind of documentation are you looking for?

**Comment by brettkail-wk** [comment]
To integrate with existing build scripts, it's easiest if I can `FROM ... as frugal` and `COPY --from=frugal /bin/frugal bin/`, so I meant documenting that the resulting binary is `/bin/frugal`.  If you'd rather not document that, that's fine.

**Comment by tylerstapler-wf** [comment]
I think I misunderstood the use case for the container. I'll find a good place for the documentation and put something together.

**Comment by brettkail-wk** [comment]
@tylerstapler-wf I also wasn't very clear about my use-case for it.  Being able to `docker run -it --rm ...` is certainly useful.

**Comment by brianshannan-wf** [comment]
+1

**Pull request merged**